src/Compiler/Checking/CheckDeclarations.fs

```diff
@@ -3945,7 +3945,21 @@ module TcDeclarations =
              | SynMemberDefn.NestedType (range=m) :: _ -> errorR(Error(FSComp.SR.tcTypesCannotContainNestedTypes(), m))
              | _ -> ()
         | ds ->
-            // Classic class construction 
+            // Check for duplicated parameters
+            for slot in ds do
+                if isAbstractSlot slot then
+                    match slot with
+                    | SynMemberDefn.AbstractSlot (x, _, m) ->
```

Comment:
Is this only checking implementation files? What about signature files?

Reply:
In my last commit I have added some logic account for this . But Im not sure if Im going at the right phase/place in the ```CheckDeclaration.fs```. Let me know if this changes are Ok please :)

Reply:
I have added more unit test to check both :)